module13and14/LoopCanvas/LoopCanvas/Views/CanvasView.swift

```diff
@@ -0,0 +1,260 @@
+//
+//  CanvasView.swift
+//  LoopCanvas
+//
+//  Created by Peter Rice on 5/30/24.
+//
+
+import SwiftUI
+import ImageIO
+
+
+struct CanvasView: View {
+  @StateObject var viewModel: CanvasViewModel
+  @State var showingRenameSongView = false
+
+  var canvasBlocksView: some View {
+    CanvasBlocksView(viewModel: viewModel)
+  }
+
+  var body: some View {
+    NavigationView {
+      ZStack {
+        ScrollView([.horizontal, .vertical]) {
+          ZStack {
+            BackgroundDots()
+
+            canvasBlocksView
+
+            GeometryReader { proxy in
+              let xOffset = proxy.frame(in: .named("CanvasCoordinateSpace")).minX
+              let yOffset = proxy.frame(in: .named("CanvasCoordinateSpace")).minY
```

Comment:
proxy.frame(in: .named("CanvasCoordinateSpace")).origin would give you a CGPoint that I think would be equal to your xOffset,yOffset

Reply:
Yup this seems to work too.